app/src/main/java/org/mozilla/focus/search/SearchEngineManager.java

```diff
@@ -34,6 +34,12 @@
 
     private List<SearchEngine> searchEngines;
 
+    /**
+     * A flag indicating that data has been loaded, or is loading. This lets us detect if data
+     * has been requested without a preceeding init().
+     */
+    private volatile boolean loadHasBeenTriggered = false;
```

Comment:
This is only read or written from synchronized blocks - so `volatile` isn't really needed. Inside synchronized blocks you'll never see stalled data.